sdk/mobile-center-push/src/main/java/com/microsoft/azure/mobile/push/Push.java

```diff
@@ -111,6 +117,29 @@ public static void setEnabled(boolean enabled) {
     }
 
     /**
+     * Enable firebase analytics collection.
+     *
+     * @param context the context to retrieve FirebaseAnalytics instance.
+     */
+    @SuppressWarnings("WeakerAccess")
+    public static void enableFirebaseAnalytics(@NonNull Context context) {
+        MobileCenterLog.debug(LOG_TAG, "Enabling firebase analytics collection.");
+        setFirebaseAnalyticsEnabled(context, true);
+    }
+
+    /**
+     * Enable or disable firebase analytics collection.
+     *
+     * @param context the context to retrieve FirebaseAnalytics instance.
+     * @param enabled <code>true</code> to enable, <code>false</code> to disable.
+     */
+    @SuppressWarnings("MissingPermission")
+    private static void setFirebaseAnalyticsEnabled(@NonNull Context context, boolean enabled) {
+        FirebaseAnalyticsUtils.setEnabled(context, enabled);
+        PreferencesStorage.putBoolean(PREFERENCE_KEY_ANALYTICS_ENABLED, enabled);
```

Comment:
Don't use storage, just a boolean field, the decision to enable firebase can be made at each run and people will call the method before start (usually right before start).

Sorry if the spec was not clear we should have clarified that it's not like a setEnabled thing like in modules.